libraries/chain/block_header_state.cpp

```diff
@@ -26,36 +26,39 @@ struct base_and_active_finalizer_policy_digest_data_t {
    digest_type base_digest;
 };
 
-// data for base_digest
-struct base_digest_data_t {
-   block_header                             header;
-   finality_core                            core;
-   flat_map<uint32_t, finalizer_policy_ptr> finalizer_policies;
-   proposer_policy_ptr                      active_proposer_policy;
-   flat_map<block_timestamp_type, proposer_policy_ptr>  proposer_policies;
-   protocol_feature_activation_set_ptr      activated_protocol_features;
-};
+// compute base_digest explicitly because of pointers involved.
+digest_type block_header_state::compute_base_digest() const {
+   digest_type::encoder enc;
+
+   fc::raw::pack( enc, header );
+   fc::raw::pack( enc, core );
+
+   for (const auto& fp_pair : finalizer_policies) {
+      fc::raw::pack( enc, *fp_pair.second );
+   }
```

Comment:
Can you not just pack the `finalizer_policies`?

Reply:
Never mind, I understand why you do not.

However, we should also pack the block number `fp_pair.first` before packing the finalizer policy.